### ISSUE TITLE
Settings menu items are empty if name contains spaces

### ISSUE THREAD

**Issue by MarcoViglione-Laser** [OrchardCMS/Orchard#7371]
While I was migrating some modules I developed to Orchard 1.10.1 I encountered a problem wherever I created a new settings menu item. If the `LocalizedString` I passed to the `GroupInfo` constructor contained a space or a special character (i.e. `context.Metadata.EditorGroupInfo.Add(new GroupInfo(T("My Settings")))`) the menu item was correctly shown but, after clicking it, it opened only an empty page.
Just to know if I need to correct my code, is this a desired behavior or a bug?

**Comment by rtpHarry** [comment]
It should still display the page. In 1.10.1 for example Audit Trail has a space, but the settings page still works.

I fixed this badly generated route issue in a PR which will be in 1.10.2:

  - https://github.com/OrchardCMS/Orchard/pull/7275/files

Try making that simple code change to your copy of Orchard and see if it fixes it?

**Comment by MarcoViglione-Laser** [comment]
I am already using your fix because I am working on the 1.10.x branch. I tried to enable the Audit Trail module and I have the same problem, as you can see from this screenshot:

![audit](https://cloud.githubusercontent.com/assets/13333789/19937429/84c3b6e2-a121-11e6-85e0-817f009e2804.png)

**Comment by rtpHarry** [comment]
oh dear, maybe my fix introduced a bug then? If you change `AuditTrail` to `Audit%20Trail` in the url does the page display correctly?

**Comment by MarcoViglione-Laser** [comment]
Unfortunately no, it gives the "Not found" error.

**Comment by rtpHarry** [comment]
I dont have time to test it at the moment but does it do with an out of the box install of Orchard 1.10.x?

Have you got some translation features enabled?

**Comment by MarcoViglione-Laser** [comment]
I tried to test it by creating a new tenant. I used the Default recipe and after the setup finished I enabled only the Audit Trail module. The problem was exactly like in the previous screenshot.

**Comment by rtpHarry** [comment]
if you comment out [my fix](https://github.com/OrchardCMS/Orchard/pull/7275/files) do the pages start working again?

**Comment by MarcoViglione-Laser** [comment]
Yes, if I comment out your fix the pages start working again. I don't know why you made that fix, but if it solved another issue maybe the right thing to do is to use the same approach wherever a groupId is used as a parameter?
I found a couple of examples:
- `public ContentShapeResult OnGroup(string groupId)` in `\Orchard\ContentManagement\Drivers\ContentShapeResult.cs`
- `public TemplateFilterForPart(string prefix, string templateName, string groupId)` and `public TemplateFilterForPart<TPart> Group(string groupId)` in `\Orchard\ContentManagement\Handlers\TemplateFilterForPart.cs`
I tried to modify the content of the first method as `_groupId = groupId.ToSafeName();` and it seems to work, but I lack a complete view of the Orchard core, so I don't know how many others methods use a groupId as a parameter or what consequences such a fix can have on the rest of the core.
